File: frontend/src/app/core/models/github/github-repository.ts
import { GithubUser } from "@core/models/github/github-user";
import {Language} from "@core/enums/language";

export interface GithubRepository {
    name: string;
    url: string;
    owner: GithubUser;
    language: Language;
    stars: number;
    isPrivate: boolean;
    description?: string;
}

File: backend/Codi.Core/Codi.Core.Common/DTO/Github/GithubRepository.cs
namespace Codi.Core.Common.DTO.Github;

public class GithubRepository
{
    public string Name { get; set; } = default!;
    public string Url { get; set; } = default!;
    public GithubUser Owner { get; set; } = default!;
    public string Language { get; set;} = default!;
    public uint Stars { get; set; } = default!;
    public bool IsPrivate { get; set; } = default!;
    public string? Description { get; set; }
}